File: src/main/java/datalake/kinesisclient/KinesisClient.java
package datalake.kinesisclient;

import java.util.concurrent.CompletableFuture;
import org.apache.logging.log4j.core.Logger;

import datalake.schema.ActivityOuterClass;
import software.amazon.awssdk.services.kinesis.*;
import software.amazon.awssdk.services.kinesis.model.*;
import software.amazon.awssdk.core.SdkBytes;

public class KinesisClient implements Activity {
    private final KinesisAsyncClient kinesis;
    private final String streamName;
    private Logger logger;

    public KinesisClient(KinesisAsyncClient kinesis, String streamName, Logger logger) {
        this.kinesis = kinesis;
        this.streamName = streamName;
        this.logger = logger;
    }

    public void recordAsync(ActivityOuterClass.Activity record, String partitionKey) {
        PutRecordRequest putRecordRequest = PutRecordRequest.builder()
                .streamName(streamName)
                .partitionKey("partition-key-1")
                .data(SdkBytes.fromByteArray(record.toByteArray()))
                .build();
        PutRecordResponse response  = kinesis.putRecord(putRecordRequest).join();
        logger.debug("ShardId" + response.shardId());
    }
}

File: src/main/csharp/datalake/kinesisClient/kinesisClient.cs
using Amazon.Kinesis;
using Amazon.Kinesis.Model;
using System;
using System.IO;
using System.Threading.Tasks;
using Datalake.Schema;
using Google.Protobuf;

namespace Datalake.kinesisClient
{
    public interface IActivity
    {
        Task RecordAsync(Activity record, string partitionKey);
    }
    public class KinesisClient : IActivity
    {
        private readonly IAmazonKinesis _kinesis;
        private readonly string _streamName;
        public KinesisClient(IAmazonKinesis kinesis, string streamName)
        {
            if (string.IsNullOrEmpty(streamName))
                throw new ArgumentNullException(nameof(streamName));
            _kinesis = kinesis;
            _streamName = streamName;
        }

        public async Task RecordAsync(Activity record, string partitionKey)
        {

            var bytes = Serialize(record);
            using (var ms = new MemoryStream(bytes))
            {
                var putRecordRequest = new PutRecordRequest
                {
                    StreamName = _streamName,
                    Data = ms,
                    PartitionKey = partitionKey
                };
                await _kinesis.PutRecordAsync(putRecordRequest);
            }
        }
        internal Byte[] Serialize(Activity record)
        {
            using (var ms = new MemoryStream())
            {
                record.WriteTo(ms);
                return ms.ToArray();
            }
        }
    }
}
